[15.0][MIG] account_chart_update

flachica opened the pull request.

percevaq committed:
[MIG]: account_char_update: Migration to version 7.0

[FIX]: Changed types to orm.Model, orm.TransientModel and orm.AbstractModel.
[FIX]: Fix deletion method for some fields of objects.
[FIX]: Contributions have been written in the standard format of the community.
[FIX]: Changes in the form to work as a real wizard.
[FIX]: Remove 'init_xml' keys, because it's no longer needed in v7.
[FIX]: Rename 'demo_xml' key to the new standard 'demo'.
[FIX]: Change imports calls.
[FIX]: Remove __author__ variables in files, because authors are put on manifest file (__openerp__.py).
[FIX]: Update view definitions to version 7.0
[FIX]: Rename variables to OpenERP standard.
[FIX]: Increased compatibility with standard PEP8.
[IMP]:  User Invitu add the method call _reopen to keep open the wizard.
[MRG]: User Invitu add frech language.

Commit:
[IMP] account_chart_update: Menu path to avoid needing multi-company permission.

Commit:
[IMP] small improvement : calling reopen method in action_init + indentation

yann-papouin committed:
FIX: wrong dict passed in _update_accounts

sbidoul committed:
[IMP] attempt to match tax code on code if no match on name was found

Commit:
[FIX] account_chart_update: Selection of operations to do now is working.
[FIX] account_chart_update: Adapted to work with inherited account chart templates.
[FIX] account_chart_update: Remove required=True for chart_template_id on DB and put on screen to avoid error when removing chart templates.
[FIX] account_chart_update: Change method to compare taxes of fiscal positions to match with the one comparing taxes.
[FIX] account_chart_update: Change method to compare accounts of fiscal positions to match with the one comparing accounts.
[IMP] account_chart_update: Wizard now shows in tree view modification notes.
[IMP] account_chart_update: For new data to be created, it is marked on notes the reason for the creation (just for clarity).
[IMP] account_chart_update: Tabs for operations that are not going to be performed are hidden.
[IMP] account_chart_update: Some PEP8 here and there.
[IMP] account_chart_update: Enhance a little the mapping searches (there's more room to improve).
[IMP] account_chart_update: Now it takes into account name and description for mapping taxes.

nbessi committed:
Set all addons to uninstallable

Commit:
[MIG] account_chart_update: Migration to 8.0

* Move out of unported
* Set account_chart_update installable
* value_reference of ir.property is a char
* Use get_by_record method of ir.property
* number digits is returned before assignment in case of property exist but account not
* product_obj isn't used

sbidoul committed:
[FIX+IMP] account_chart_update: Several things

* search on code first to match tax codes

  The tax code code should be more discriminant that the tax code name.

* compare more fields in tax codes
* deletion of tax codes removed from templates

  For tax codes, it make sense to propose deletion, since tax codes
  are normally defined by the tax authority and the user does not
  customize them.

* update accounts on taxes even if an update of the account was not requested

  Useful when the chart of taxes changes but the chart of account
  need not be updated.

* depth-first search of tax codes

  This helps creating parents before children.

* compare more fields in taxes
* add active field on tax code

  Deactivate tax codes which are not in the template.

* deactivate taxes that are not in the template
* multi-company criteria for taxes
* avoid inserting duplicates in presence of inactive taxes or tax codes

pedrobaeza committed:
[IMP] account_chart_update: Whole refactoring

* Fix calculated field when showing inactives + add field to view
* Migrated to new API.
* Provided hooks for detecting changes and for adding update/create values.
* Purge dead error detection code (not applying).
* Use global mapping dictionaries stored on class instead of passing arguments down to top and viceversa.
* Number fields computed by a function field instead of being computed by hand.
* Use recordsets as indexes instead of ids to avoid repetitive browses.
* Re-enable local cache variables
* README + authors

Commit:
[FIX+IMP] account_chart_update: Several things

* Set updated taxes as active
* Check parent_id field in '_is_different_tax' method
* PEP8
* Spanish translation

pedrobaeza committed:
[FIX] account_chart_update: Call mapping function instead of removed cache. Closes #240

Commit:
[FIX][account-financial-tools] depends key is duplicated

pedrobaeza committed:
[FIX] account_chart_update: Don't write code on account update

As there's a constraint that forbids to write the code of the account
if it has moves, we delete the corresponding entry on update only,
because the method that prepares the values is the same for new and
existing accounts.

pablocm-aserti committed:
[FIX] Include acc templates from parent CoA templates

sbidoul committed:
[FIX] account_chart_update:

* Process price_include field
* Detect change in price_include field

yajo committed:
[MIG][9.0][account_chart_update] Migrate.

- New-style license headers.
- Remove .pot file.
- Remove tax codes stuff, now removed from v9.
- Refactor methods for search, create, update, delete. Now they are smaller, fitter, happier, more productive.
- Only update fields that have any kind of change on any updated record.
- Place the wizard in the configuration page, instead of its own menu item.
- Display amount of disabled taxes at ending page.

Commit:
[MIG] account_chart_update: Migration to 10.0

pedrobaeza committed:
[FIX+IMP] account_chart_update: Fix fiscal position mapping + translations + clean authors

As this module has been written from scratch with a totally different technique, I clean the list and update copyrights.

pedrobaeza committed:
[FIX] account_chart_update: Fetch account code digits correctly

amcor committed:
[MIG][ADD] i18n directly from 10.0

amcor committed:
[MIG] account_chart_update: Migration to 11.0

sbidoul committed:
account_chart_update: copy inactive tax templates too

In v10, tax templates have an active flag. It is useful to copy them too, letting the user select which one he wants to activate in it's instance.

MouTio committed:
[FIX] account_chart_update: Limit fiscal position search with same name (#651)

pedrobaeza committed:
[FIX+IMP] account_chart_update: Major refactoring

* Compare all fields from template

  Not all fields were properly compared with previous code. This means also
  to exclude some of them, but now we have a full and extensible system.

* Optimize cached method
* Generate accounts and fiscal positions using Odoo methods (tax already did)
* README by fragments
* Add tests up to full coverage

pedrobaeza committed:
[FIX] account_chart_update: Don't cache recordsets

When marking methods as ormcache, you can't return recordsets, as they keep
the cursor information, and that cursor can be closed (worker spawn for example)
when you need to access again data.

This is fixed returning IDs instead, and browsing when needed

pedrobaeza committed:
[MIG] account_chart_update: Additional needed changes

pedrobaeza committed:
[FIX] account_chart_update: Handle updated m2m correctly

pedrobaeza committed:
[IMP] account_chart_update: Include field selection

With this option, you can select which fields do you want to compare for updating.

Use case: you have specific accounts in taxes, and you don't want to lose them,
but you want to update their groups through the wizard. Before this, the update
was all or nothing. Now, you have total control!

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

ernestotejeda committed:
account_chart_update: allows to set the matching value order

It allows to set the order of matching for Account, Taxes and
Fiscal position.

tv-openbig committed:
Translated using Weblate (German)

Currently translated at 100.0% (100 of 100 strings)

Translation: account-financial-tools-11.0/account-financial-tools-11.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-11-0/account-financial-tools-11-0-account_chart_update/de/

ernestotejeda committed:
account_chart_update: Fix number of updates and insertions in log

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-11.0/account-financial-tools-11.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-11-0/account-financial-tools-11-0-account_chart_update/

ernestotejeda committed:
[MIG] account_chart_update: Migration to 12.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 72.0% (85 of 118 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/es/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 30.5% (36 of 118 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_chart_update/hr/

ernestotejeda committed:
[IMP] account_chart_update: black, isort

ernestotejeda committed:
[MIG] account_chart_update: Migration to 13.0

pedrobaeza committed:
[FIX] account_chart_update: Clear caches before searching

If not, changes done in for example taxes are not reflected.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_chart_update/

OCA-git-bot committed:
pre-commit update

pedrobaeza committed:
[FIX+IMP] account_chart_update: Create taxes in batch

With this, we gain some performance, but also prevent an error if there are template
taxes that have children taxes, as Odoo standard method forces that in the same call
all of them are created simultaneously, or it will enter in an infinite loop.

pedrobaeza committed:
[FIX+IMP] account_chart_update: Add consistency method

With this, we prevent lock due to infinite loop if children taxes are matched, but
not the parent one and it's marked to be created.

ps-tubtim committed:
[13.0][FIX] account_chart_update: Remove @api.multi

OCA-git-bot committed:
account_chart_update 13.0.1.0.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_chart_update/

pedrobaeza committed:
[IMP] account_chart_update: Search for real XML-IDs only

Previously, only a check on an existing XML-ID which is not coming from exportation
was done, but now we check that the XML-ID is the exact one. With this, we avoid
that the same record changed their XML-ID and thus, there's no more link between
them.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_chart_update/

luc-demeyer committed:
[13.0][FIX]account_chart_update inactive taxes false positives

luc-demeyer committed:
[13.0][FIX]account_chart_update refactor tax update

OCA-git-bot committed:
account_chart_update 13.0.1.0.5

luc-demeyer committed:
[FIX] account_chart_update: ignore template fields that do not exist in real

Reyes4711-S73 committed:
[13.0][IMP] account_chart_update: black, isort, prettier

Reyes4711-S73 committed:
[14.0][MIG] account_chart_update: Migration to 14.0

DonatasV committed:
[IMP] account_chart_update: update tests

pedrobaeza committed:
[FIX] account_chart_update: cash_basis_base_account_id no longer exists

This field disappears in this version.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_chart_update
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_chart_update/

MiquelRForgeFlow committed:
[FIX] account_chart_update: Warning -> UserError

MiquelRForgeFlow committed:
[FIX] account_loan: post() -> action_post()

OCA-git-bot committed:
account_chart_update 14.0.2.0.2

flachica committed:
[IMP] account_chart_update: black, isort, prettier

flachica committed:
Standard migration

flachica committed:
Migration and some fixes:

   * Ignore changes in mail thread fields
   * account.tax.template._generate_tax return dict of objects
     and wizard fails
   * Load generic coa localization module to improve the test

MiquelRForgeFlow commented:
@flachica Currently, account_asset_management is in 14.0.2.6.1 version, so I think you should update the commit history of this PR accordingly 😉

flachica commented:
After these Odoo [changes](https://github.com/odoo/odoo/commit/5419fe445dbaab24bdda1e7a5d25e4fac3fb2f1c) the tests is failing

I'm change [this](https://github.com/flachica/account-financial-tools/blob/15.0-mig-account_chart_update/account_chart_update/tests/test_account_chart_update.py#L93:L112) point but is not enought and I don't know if I'm in the correct way

Basically I think it boils down to the fact that now we do depend on having certain parts of a generic localization package installed. But I would like to confirm that I am correct. But it seems so far-fetched to me that I suppose there will be some simpler solution

Any help is welcome

rafaelbn commented:
/ocabot migration account_chart_update

LoisRForgeFlow committed:
[FIX] account_chart_update: do not match twice the same repartition line

When updating a tax with several repartition lines, the same existing
candidates were considered in all iteration which could lead the wizard
to match the same repartition line with different lines of the template
and not detecting any change to do.

MiquelRForgeFlow commented:
@flachica please, cherry pick https://github.com/OCA/account-financial-tools/commit/0ac8e6045aa507d8d501c7828357d8411d27cc52

flachica commented:
> @flachica please, cherry pick [0ac8e60](https://github.com/OCA/account-financial-tools/commit/0ac8e6045aa507d8d501c7828357d8411d27cc52)

Done

Shide commented:
I found a functional error:
Steps to reproduce:
1. Start a clean database
2. Set company country to Spain
3. Install 10n_es
4. Install account_chart_update
5. Run straight account chart update
6. Clic next
7. Error happens:
```
One or more errors detected!

Updated account '100000 - Capital social'.
Updated account '100000 - Capital social'.
Updated account '103000 - Socios por desembolsos no exigidos, capital social'.
Updated account '103000 - Socios por desembolsos no exigidos, capital social'.
ERROR: Exception writing account '103000 - Socios por desembolsos no exigidos, capital social'.
Traceback (most recent call last):
  File "/opt/odoo/auto/addons/account_chart_update/wizard/wizard_chart_update.py", line 1026, in _update_accounts
    _logger.info(
  File "/usr/local/lib/python3.8/contextlib.py", line 120, in __exit__
    next(self.gen)
  File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 135, in savepoint
    self.flush()
  File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 108, in flush
    self.transaction.flush()
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 833, in flush
    env_to_flush['base'].flush()
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5644, in flush
    process(self.env[model_name], id_vals)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5635, in process
    recs._write(vals)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3949, in _write
    cr.execute(query, params + [sub_ids])
  File "<decorator-gen-3>", line 2, in execute
  File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 89, in check
    return f(self, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 310, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "mail_followers_mail_followers_res_partner_res_model_id_uniq"
DETAIL:  Key (res_model, res_id, partner_id)=(account.account, 0, 2) already exists.
```

Shide commented on account_chart_update/wizard/wizard_chart_update.py:
```diff
@@ -0,0 +1,1320 @@
+            "account.account.template": {"chart_template_id", "root_id", "nocreate"},
+            "account.fiscal.position.template": {"chart_template_id"},
+        }
+        specials = {
```
```suggestion
        # Exclude `mail.thread` fields
        specials = set(self.env['mail.thread']._fields) | {
```

I've seen the problem is when trying to replace data on the `mail.message` fields.

I suggest to add all fields of the `mail.thread` model onto the fields to ignore. 
In previous versions, `account.account.template` doesn't inherit from `mail.thread`, and these fields makes not sense to override existing data. Maybe someone put some important messages on this model and we don't want to overwrite them.

flachica replied:
Done

yajo reviewed (approved):
code review

yajo commented:
The fix for the broken build is in https://github.com/OCA/account-financial-tools/pull/1360.

rafaelbn reviewed (approved):
Thanks 👍

rafaelbn commented:
/ocabot rebase

rafaelbn commented:
/ocabot merge patch

rafaelbn commented:
Please @flachica could you rebase? Because https://github.com/OCA/account-financial-tools/pull/1346#issuecomment-1084210780

rafaelbn commented:
Please @flachica could you rebase?

flachica commented:
> Please @flachica could you rebase?

Done

rafaelbn commented:
/ocabot merge nobump

rafaelbn commented:
Thank you @flachica , still failing test. Please check it if possible! Thank you very mucho for your contributions!

😄 

```
2022-04-04 11:28:40,516 257 ERROR odoo odoo.addons.account_chart_update.tests.test_account_chart_update: ERROR: TestAccountChartUpdate.test_chart_update
Traceback (most recent call last):
  File "/__w/account-financial-tools/account-financial-tools/account_chart_update/tests/test_account_chart_update.py", line 131, in setUp
    chart_by_company_user.try_loading()
  File "/opt/odoo/addons/account/models/chart_template.py", line 198, in try_loading
    template.with_context(default_company_id=company.id)._load(15.0, 15.0, company)
  File "/opt/odoo/addons/account/models/chart_template.py", line 283, in _load
    acc_template_ref, taxes_ref = self._install_template(company, code_digits=self.code_digits)
  File "/opt/odoo/addons/account/models/chart_template.py", line 568, in _install_template
    tmp1, tmp2 = self.with_context(lang=company.partner_id.lang)._load_template(company, code_digits=code_digits, account_ref=acc_ref, taxes_ref=taxes_ref)
  File "/opt/odoo/addons/account/models/chart_template.py", line 596, in _load_template
    generated_tax_res = self.with_context(active_test=False).tax_template_ids._generate_tax(company)
  File "/opt/odoo/addons/account/models/chart_template.py", line 1151, in _generate_tax
    raise UserError(_("Please first define a fiscal country for company %s.", company.name))
odoo.exceptions.UserError: Please first define a fiscal country for company Test account_chart_update company.
 
2022-04-04 11:28:40,516 257 INFO odoo odoo.addons.account_chart_update.tests.test_account_chart_update: Starting TestAccountChartUpdate.test_matching ... 
2022-04-04 11:28:40,989 257 INFO odoo odoo.addons.account_chart_update.tests.test_account_chart_update: ====================================================================== 
2022-04-04 11:28:40,989 257 ERROR odoo odoo.addons.account_chart_update.tests.test_account_chart_update: ERROR: TestAccountChartUpdate.test_matching
Traceback (most recent call last):
  File "/__w/account-financial-tools/account-financial-tools/account_chart_update/tests/test_account_chart_update.py", line 131, in setUp
    chart_by_company_user.try_loading()
  File "/opt/odoo/addons/account/models/chart_template.py", line 198, in try_loading
    template.with_context(default_company_id=company.id)._load(15.0, 15.0, company)
  File "/opt/odoo/addons/account/models/chart_template.py", line 283, in _load
    acc_template_ref, taxes_ref = self._install_template(company, code_digits=self.code_digits)
  File "/opt/odoo/addons/account/models/chart_template.py", line 568, in _install_template
    tmp1, tmp2 = self.with_context(lang=company.partner_id.lang)._load_template(company, code_digits=code_digits, account_ref=acc_ref, taxes_ref=taxes_ref)
  File "/opt/odoo/addons/account/models/chart_template.py", line 596, in _load_template
    generated_tax_res = self.with_context(active_test=False).tax_template_ids._generate_tax(company)
  File "/opt/odoo/addons/account/models/chart_template.py", line 1151, in _generate_tax
    raise UserError(_("Please first define a fiscal country for company %s.", company.name))
odoo.exceptions.UserError: Please first define a fiscal country for company Test account_chart_update company.
```

Shide commented on account_chart_update/wizard/wizard_chart_update.py:
```diff
@@ -0,0 +1,1321 @@
+        :param str name: The name of the template model.
+        :return set: Fields to ignore in diff.
+        """
+        specials_mapping = specials = {
```
Why add `specials_mapping = specials`? You are overwritting it on L667

flachica replied:
It is a typo. I push new code to solve it but tests still having problems. I need more time to try to solve it

Shide commented:
I got an error:

Steps:
- Tested on Runboat
- Install l10n_es
- Duplicate 400000 and 600000 to 499999 and 699999 accounts and set it to Customer Invoices Journals and Vendor Bills Journals.
- Run wizard for PGCE PYMEs 2008 with Recreate missing XML-IDs

```
RPC_ERROR
Odoo Server Error


Traceback (most recent call last):
  File "/opt/odoo/odoo/tools/cache.py", line 85, in lookup
    r = d[key]
  File "/opt/odoo/odoo/tools/func.py", line 71, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/odoo/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('wizard.update.charts.accounts', <function WizardUpdateChartsAccounts.find_account_by_templates at 0x7f1c0900fc10>, account.account.template(account.account.template(275,),))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/odoo/api.py", line 886, in get
    return field_cache[record._ids[0]]
KeyError: account.account.template(275,)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/odoo/fields.py", line 1057, in __get__
    value = env.cache.get(record, self)
  File "/opt/odoo/odoo/api.py", line 889, in get
    raise CacheMiss(record, field)
odoo.exceptions.CacheMiss: 'account.account.template(account.account.template(275,),).code'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/odoo/http.py", line 688, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo/http.py", line 360, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/http.py", line 349, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo/http.py", line 917, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo/http.py", line 536, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/addons/web/controllers/main.py", line 1352, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/main.py", line 1340, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 464, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 451, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/mnt/data/odoo-addons-dir/account_chart_update/wizard/wizard_chart_update.py", line 374, in action_update_records
    self._update_taxes_pending_for_accounts(todo_dict)
  File "/mnt/data/odoo-addons-dir/account_chart_update/wizard/wizard_chart_update.py", line 1077, in _update_taxes_pending_for_accounts
    acc_id = self.find_account_by_templates(
  File "<decorator-gen-209>", line 2, in find_account_by_templates
  File "/opt/odoo/odoo/tools/cache.py", line 90, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/mnt/data/odoo-addons-dir/account_chart_update/wizard/wizard_chart_update.py", line 534, in find_account_by_templates
    codes = templates.mapped("code")
  File "/opt/odoo/odoo/models.py", line 5447, in mapped
    recs = recs._fields[name].mapped(recs)
  File "/opt/odoo/odoo/fields.py", line 1176, in mapped
    self.__get__(first(remaining), type(remaining))
  File "/opt/odoo/odoo/fields.py", line 1083, in __get__
    recs._fetch_field(self)
  File "/opt/odoo/odoo/models.py", line 3276, in _fetch_field
    self._read(fnames)
  File "/opt/odoo/odoo/models.py", line 3343, in _read
    cr.execute(query_str, params + [sub_ids])
  File "<decorator-gen-3>", line 2, in execute
  File "/opt/odoo/odoo/sql_db.py", line 89, in check
    return f(self, *args, **kwargs)
  File "/opt/odoo/odoo/sql_db.py", line 310, in execute
    res = self._obj.execute(query, params)
  File "/opt/odoo-venv/lib/python3.8/site-packages/psycopg2/extensions.py", line 121, in getquoted
    pobjs = [adapt(o) for o in self._seq]
  File "/opt/odoo-venv/lib/python3.8/site-packages/psycopg2/extensions.py", line 121, in <listcomp>
    pobjs = [adapt(o) for o in self._seq]
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 644, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 302, in _handle_exception
    raise exception.with_traceback(None) from new_cause
psycopg2.ProgrammingError: can't adapt type 'account.account.template'
```

flachica commented:
> * set it to Customer Invoices and Vendor Bills.

Sorry for this newbie question, but how I set accounts to Customer Invoices and Vendor Bills?

Shide commented:
> > * set it to Customer Invoices and Vendor Bills.
> 
> Sorry for this newbie question, but how I set accounts to Customer Invoices and Vendor Bills?

Sorry, i forgot to add "Journals" word at the end.

On the Journals, change the main accounts to other than 400000 and 600000

flachica commented:
> * 699999

I could not replicate the error in local environment. But once I read the code, make some changes like [this](https://github.com/flachica/account-financial-tools/blob/cb3251041301ac1cf02358bb01d1b1f1a96fe8d2/account_chart_update/wizard/wizard_chart_update.py#L1078). I added .id to correct browse record.

rafaelbn commented:
- Customer Journals usually should have income accounts, in Spain something like 700000 or 705000
- Vendor Journals usually should have income accounts, in Spain something like 600000 or 629000

But in real life the accounts in customer or vendor journals are not used generally because the accounts come from the products (or product category) in the invoice lines.

manuelregidor commented:
HI @flachica Do you think you'll be able to continue with this PR? Thank you.

rafaelbn commented:
Hello @manuelregidor if we have 1 more reivew like you or @HaraldPanten we can merge this PR as is 😄 💪 Thank you!

HaraldPanten reviewed (approved):
It did work well for me in runbot.

LGTM

rafaelbn commented:
/ocabot merge patch
